Weather_App/Weather_App/Scenes/Map/MapViewController.swift

```diff
@@ -103,5 +107,76 @@ extension MapViewController {
         if let icon = weatherCurrent.weatherStatus {
             statusImageView.loadImage(withIcon: icon)
         }
+        updateFavoriteStatus(for: weatherCurrent.nameCity)
+    }
+}
+
+extension MapViewController {
+    @IBAction private func favoriteButtonTapped(_ sender: Any) {
+        guard let weatherData = currentWeatherData else { return }
+        if isFavorite {
+            if let weatherEntity = fetchWeatherEntity(for: weatherData.nameCity) {
+                deleteWeatherFromeCoreData(weatherEntity: weatherEntity)
+                isFavorite = false
+                updateFavoriteButtonImage()
+            }
+        } else {
+            saveWeatherToCoreData(
+                latitude: weatherData.coordinate.latitude,
+                longitude: weatherData.coordinate.longitude,
+                cityName: weatherData.nameCity)
+            isFavorite = true
+            updateFavoriteButtonImage()
+        }
+    }
+    
+    private func updateFavoriteButtonImage() {
+        let imageName = isFavorite ? "iconStarYellow" : "iconStar"
+        let image = UIImage(named: imageName)
+        favoriteButton.setImage(image, for: .normal)
+    }
+    
+    private func updateFavoriteStatus(for cityName: String) {
+        if let weatherEntity = fetchWeatherEntity(for: cityName) {
+            isFavorite = true
+        } else {
+            isFavorite = false
+        }
```

Comment:
thử cái này coi sao e
```suggestion
        isFavorite = fetchWeatherEntity(for: cityName) != nil
```